source/Meadow-Endpoints.js

```diff
@@ -435,8 +435,14 @@ var MeadowEndpoints = function()
 				return tmpCallback('Endpoint \'' + pMethod + '\' does not exist!'); //might be better as an exception
 			}
 
-			//TODO: should switch depending on type
+			// TODO: should switch depending on type
+			// TODO: should we keep this around, just make a deep copy of 'pOptions'
 			var pRequest = {params: pData, formattedParams: pData, body: pData};
+			if (typeof(pOptions) === 'object') {
+				// copy header (function) and headers (raw headers)
+				pRequest.header = pOptions.header;
+				pRequest.headers = pOptions.headers;
```

Comment:
Do these need a `bind` call, or do they work independent of their hosting object?

Reply:
they actually work! The `header` function only depends on the `headers` map, so this chunk is fully independent from internal workings of restify

Reply:
Is there a risk of this behavior changing (ex. if we upgrade to a newer Restify version)?

Reply:
Said another way, If we just added `header`, but bound it, we wouldn't need to bring over `headers`, and it would be more compatible.